prow/jira/jira.go

```diff
@@ -364,17 +680,57 @@ func maskAuthorizationHeader(key string, value string) string {
 	return value
 }
 
-// JiraError collapses cryptic Jira errors to include response
+type JiraError struct {
+	StatusCode    int
+	Body          string
+	OriginalError error
+}
+
+func (e JiraError) Error() string {
+	return fmt.Sprintf("%s: %s", e.OriginalError, e.Body)
+}
+
+// JiraErrorStatusCode will identify if an error is a JiraError and return the
+// stored status code if it is; if it is not, `-1` will be returned
+func JiraErrorStatusCode(err error) int {
+	if jiraErr := (JiraError{}); errors.As(err, &jiraErr) {
```

Comment:
Shouldn't it be
```suggestion
	if jiraErr := &JiraError{}; errors.As(err, &jiraErr) {
```

Does errors.As match if the error is a pointer type and you try to match it to the corresponding value type as here?
Also, these two should have tests